wallet/how-to/connect/index.md

```diff
@@ -1,166 +1,313 @@
+// Connect to the selected provider using eth_requestAccounts.
+const connectWithProvider = async (wallet: EIP6963AnnounceProviderEvent["detail"]) => {
+    try {
+        await wallet.provider
+            .request({ method: "eth_requestAccounts" })
+    } catch (error) {
+        console.error("Failed to connect to provider:", error)
+    }
+}
 
-export const store = {
-    value: ()=>providers,
-    subscribe: (callback: ()=>void)=>{
-        function onAnnouncement(event: EIP6963AnnounceProviderEvent){
-            if(providers.map(p => p.info.uuid).includes(event.detail.info.uuid)) return
-            providers = [...providers, event.detail]
-            callback()
+// Display detected providers as connect buttons.
+export function listProviders(element: HTMLDivElement) {
+    window.addEventListener("eip6963:announceProvider",
+        (event: EIP6963AnnounceProviderEvent) => {
+            const button = document.createElement("button")
+        
+            button.innerHTML = `
+                <img src="${event.detail.info.icon}" alt="${event.detail.info.name}" />
+                <div>${event.detail.info.name}</div>
+            `
+        
+            // Call connectWithProvider when a user selects the button.
+            button.onclick = () => connectWithProvider(event.detail)
+            element.appendChild(button)
         }
-        window.addEventListener("eip6963:announceProvider", onAnnouncement);
-        window.dispatchEvent(new Event("eip6963:requestProvider"));
-    
-        return ()=>window.removeEventListener("eip6963:announceProvider", onAnnouncement)
+    )
+
+    // Notify event listeners and other parts of the dapp that a provider is requested.
+    window.dispatchEvent(new Event("eip6963:requestProvider"))
+}
+```
+
+The `connectWithProvider` function connects the user to the selected provider using
+[`eth_requestAccounts`](/wallet/reference/eth_requestaccounts).
+The `wallet` object is passed as an argument to the function, indicating the argument type.
+
+The `listProviders` function uses a simplified approach.
+Instead of mapping and joining an entire block of HTML, it directly passes the `event.detail` object
+to the `connectWithProvider` function when a provider is announced.
+
+#### 5. View the project
+
+Run the following command to view and test the Vite project in your browser:
+
+```bash
+npm run dev
+```
+
+#### Example
+
+See the [Vanilla TypeScript example](https://github.com/MetaMask/vite-vanilla-ts-eip-6963) for more information.
+You can clone the repository and run the example locally using `npm i && npm run dev`.
+
+### React TypeScript
+
+Follow these steps for creating a React TypeScript project to connect to MetaMask:
+
+#### 1. Create a project
+
+[Create a Vite project](https://v3.vitejs.dev/guide/#scaffolding-your-first-vite-project) using the
+template for React TypeScript:
+
+```bash
+npm create vite@latest react-ts-6963 -- --template react-ts
```

Comment:
Can this be more succinct or are the two sets of  `--`s needed to ? 

```suggestion
npm create vite@latest react-ts-6963 -- --template react-ts
```

Reply:
Extra -- is needed for npm 7+, as per the linked Vite docs: https://v3.vitejs.dev/guide/#scaffolding-your-first-vite-project